R/FilterStateChoices.R

```diff
@@ -355,7 +352,21 @@ ChoicesFilterState <- R6::R6Class( # nolint
           "are not in choices of column", private$varname, "in dataset", private$dataname, "."
         ))
       }
-      values[in_choices_mask]
+      values <- values[in_choices_mask]
+
+      if (is.null(values)) {
+        values <- if (private$multiple) private$choices else private$choices[1]
+      } else {
+        if (length(values) > 1 && !private$multiple) {
+          warning(paste(
+            "Values:", strtrim(paste(values, collapse = ", "), 360),
+            "are not a vector of length one.",
+            "The first value will be selected by default."
+          ))
+        }
+        values <- if (private$multiple) values else values[1]
+      }
```

Comment:
1. is.null(values) will never be TRUE because of [this](https://github.com/insightsengineering/teal.slice/blob/e61bca49b7a3f12f64519a4c22838b743a5dd2a0/R/FilterState.R#L416)
2. This means that we just need to handle the situation when `multiple = FALSE` and `length(value) != 1`

I propose for this to have set_selected in FilterStateChoices and remove all the extra code you introduced here. `FilterStateChoices$set_selected` could look like:
```r
set_selected = function(value) {
  if (length(value) != 1) {
    warning("same warning as you wrote here")
    value <- private$choices[1]
  }
  super$set_selected(value)
}
```

Reply:
This fixed. modified observeEvent for input$selection.